tools/scanner/cmd/scanner.go

```diff
@@ -0,0 +1,136 @@
+package main
+
+import (
+	"bytes"
+	"errors"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/google/osv/tools/scanner/internal/osv"
+	"github.com/google/osv/tools/scanner/internal/sbom"
+)
+
+func usage() {
+	fmt.Fprintf(flag.CommandLine.Output(), "%s [flags] dir1 dir2...\n", os.Args[0])
+	flag.PrintDefaults()
+}
+
+func scan(arg string) error {
+	info, err := os.Stat(arg)
+	if err != nil {
+		return err
+	}
+
+	if info.IsDir() {
+		return scanDir(arg)
+	}
+
+	// Try guessing file type and scanning the SBOM.
+	log.Printf("Scanning file %s\n", arg)
+	file, err := os.Open(arg)
+	if err != nil {
+		return err
+	}
+
+	scanIdentifier := func(id sbom.Identifier) error {
+		resp, err := osv.MakePURLRequest(id.PURL)
+		if err != nil {
+			return err
+		}
+		printResults(id.PURL, resp)
+		return nil
+	}
+
+	for _, provider := range sbom.Providers {
+		file.Seek(0, os.SEEK_SET)
+		err := provider.GetPackages(file, scanIdentifier)
+		if err == nil {
+			// Found the right format.
+			log.Printf("Scanned %s SBOM", provider.Name())
+			return nil
+		}
+
+		if errors.Is(err, sbom.InvalidFormat) {
+			continue
+		}
+
+		return err
+	}
```

Comment:
Moving this logic into its own function make make the code easier to read (e.g. `scanFile()`)

Reply:
Done!